client/main.lua

```diff
@@ -615,26 +658,41 @@ RegisterNetEvent('qbx_vehicleshop:client:swapVehicle', function(data)
     local dataTargetVehicle = sharedConfig.shops[shopName].showroomVehicles[data.targetVehicle]
     if dataTargetVehicle.vehicle == data.toVehicle then return end
 
-    local closestVehicle = lib.getClosestVehicle(dataTargetVehicle.coords.xyz, 5, false)
-    if not closestVehicle then return end
-
     if not IsModelInCdimage(data.toVehicle) then
         lib.print.error(('Failed to find model for "%s". Vehicle might not be streamed?'):format(data.toVehicle))
         return
     end
 
-    DeleteEntity(closestVehicle)
-    while DoesEntityExist(closestVehicle) do
-        Wait(50)
+    local points = lib.points.getAllPoints()
+    local vehPoint = nil
+    for i = 1, #points do
+        local point = points[i]
+        if point.coords == dataTargetVehicle.coords.xyz then
+            vehPoint = point
+            break
+        end
     end
```

Comment:
This search is actually error-prone because when points are removed they are `nil` (in which case `points[i]` is nil).
Instead of running a search based on the coords, save the points beforehand based on the shop name and the shop's current vehicle number.

Reply:
Refactored to use a lookup table similar to the original construct. We don't actually remove points anymore so this probably wouldn't be an issue but the new way is more performant since no array iteration or vector math is required.